Cabal-syntax/src/Distribution/Utils/Path.hs

```diff
@@ -531,3 +536,15 @@ data Response
 --
 -- See Note [Symbolic paths] in Distribution.Utils.Path.
 data PkgConf
+
+-------------------------------------------------------------------------------
+
+-- * Windows utils
+
+-------------------------------------------------------------------------------
+
+-- | Sometimes we need to represent a Windows path (that might have been
+-- normalized) as a POSIX path, for example in URIs.
+asPosixPath :: FilePath -> FilePath
+asPosixPath p =
+  [if x == Windows.pathSeparator then Posix.pathSeparator else x | x <- p]
```

Comment:
The docs don't recommend direct comparison but to use [isPathSeparator](https://hackage.haskell.org/package/filepath-1.5.4.0/docs/System-FilePath.html#v:isPathSeparator) instead.

> Rather than using (== pathSeparator), use this.

Reply:
Im fine using `isPathSeparator` but keep in mind that
```
> Windows.isPathSeparator Posix.pathSeparator
True
```

Reply:
I saw that too. I'm OK either way. Maybe a short comment attached to your choice?

Reply:
Now that `asPosixPath` is a one line list comprehension does it make sense to add it to `Cabal-syntax`? There's only two uses, in `cabal-install` and `cabal-testsuite`. Perhaps just inline it where it is used and avoid touching `Cabal-syntax`?

Reply:
I prefer to not inline it, as @ulysses4ever suggests. I agree with him.

I added a comment to the function.